internal/core/services/rooms/status.go

```diff
@@ -72,10 +97,32 @@ func (sm *SingleMatchStatusCalculator) CalculateRoomStatus(room game_room.GameRo
 
 type MultipleMatchStatusCalculator struct {
 	scheduler entities.Scheduler
+	config    RoomManagerConfig
 	logger    *zap.Logger
 }
 
 func (mm *MultipleMatchStatusCalculator) CalculateRoomStatus(room game_room.GameRoom, instance game_room.Instance) (game_room.GameRoomStatus, error) {
+	// Check if room is allocated and within TTL
+	if room.Status == game_room.GameStatusAllocated && room.AllocatedAt != nil {
+		timeSinceAllocation := time.Since(*room.AllocatedAt)
+		if timeSinceAllocation <= mm.config.RoomAllocationTTL {
+			// Room is still within allocation TTL
+			composedStatus, err := room.RoomComposedStatus(instance.Status.Type)
+			if err != nil {
+				return room.Status, fmt.Errorf("error calculating room composed status: %w", err)
+			}
+
+			// Allow transition to Occupied if room reports occupied
+			if composedStatus == game_room.GameStatusOccupied {
+				return game_room.GameStatusOccupied, nil
+			}
```

Comment:
Shouldn't we treat game_room.GameStatusActive as well? Or does it either transition to active / occupied in the same code? Isn't Maestro the one that decides that the room is occupied with current implementation?